tests/curators/test_curators_general.py

```diff
@@ -242,6 +242,24 @@ def test_schema_not_saved(df):
     )
 
 
+def test_schema_artifact_annotated(df):
+    """A passed Artifact should be annotated with a Schema if successfully curated."""
+    af = ln.Artifact.from_df(df, key="test.parquet").save()
+    schema = ln.Schema(
+        name="sample schema",
+        features=[ln.Feature(name="sample_id", dtype="str").save()],
+    ).save()
+    curator = ln.curators.DataFrameCurator(af, schema)
+    curator.validate()
+    curator.save_artifact()
+    assert af.schema is not None
```

Comment:
this here doesn't do proper check of the FK in the DB. The artifact needs to be queried from the DB first before checking.